contentcuration/contentcuration/models.py

```diff
@@ -1381,102 +1381,232 @@ def get_nodes_with_title(cls, title, limit_to_children_of=None):
-        max_level = max(resources.values_list('level', flat=True).distinct() or [0])
-        deepest_node = resources.filter(level=max_level).first()
-        pathway = list(deepest_node.get_ancestors()
-                       .exclude(parent=None)
-                       .values('title', 'node_id', 'kind_id')
-                       ) if deepest_node else []
-        sample_nodes = [
-            {
-                "node_id": n.node_id,
-                "title": n.title,
-                "description": n.description,
-                "thumbnail": n.get_thumbnail(),
-                "kind": n.kind_id,
-            } for n in deepest_node.get_siblings(include_self=True)[0:4]
-        ] if deepest_node else []
+        max_level = max(
+            resources.values_list("level", flat=True).order_by().distinct() or [0]
+        )
+        m_nodes = With(
+            resources.values("id", "level", "tree_id", "lft").order_by(),
+            name="m_nodes",
+        )
+        depeest_node_record = (
+            m_nodes.queryset()
+            .with_cte(m_nodes)
+            .filter(level=max_level)
+            .values("id")
+            .order_by("tree_id", "lft")
+            .first()
+        )
+        if depeest_node_record:
+            deepest_node = ContentNode.objects.get(pk=depeest_node_record["id"])
+        pathway = (
+            list(
+                deepest_node.get_ancestors()
+                .order_by()
+                .exclude(parent=None)
+                .values("title", "node_id", "kind_id")
+                .order_by()
+            )
+            if depeest_node_record
+            else []
+        )
+        sample_nodes = (
+            [
+                {
+                    "node_id": n.node_id,
+                    "title": n.title,
+                    "description": n.description,
+                    "thumbnail": n.get_thumbnail(),
+                    "kind": n.kind_id,
+                }
+                for n in deepest_node.get_siblings(include_self=True)[0:4]
+            ]
+            if depeest_node_record
+            else []
+        )
 
         # Get list of channels nodes were originally imported from (omitting the current channel)
         channel_id = channel and channel.id
-        originals = resources.values("original_channel_id") \
-            .annotate(count=Count("original_channel_id")) \
+        originals = (
+            resources.values("original_channel_id")
+            .annotate(count=Count("original_channel_id"))
             .order_by("original_channel_id")
-        originals = {c['original_channel_id']: c['count'] for c in originals}
-        original_channels = Channel.objects.exclude(pk=channel_id) \
+        )
+        originals = {c["original_channel_id"]: c["count"] for c in originals}
+        original_channels = (
+            Channel.objects.exclude(pk=channel_id)
             .filter(pk__in=[k for k, v in list(originals.items())], deleted=False)
```

Comment:
Maybe change this to `originals.keys()` instead of this very curious list comprehension?

Reply:
lol, sure, I had not seen it as I focused only in the sql optimizations